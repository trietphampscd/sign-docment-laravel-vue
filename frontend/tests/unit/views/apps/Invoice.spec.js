import Vue from 'vue'
import { shallowMount, mount } from '@vue/test-utils'
import BootstrapVue from 'bootstrap-vue'
import Invoice from '@/views/apps/invoicing/Invoice'

Vue.use(BootstrapVue)

describe('Invoice.vue', () => {
  it('has a name', () => {
    expect(Invoice.name).toMatch('invoice')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Invoice)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Invoice', () => {
    const wrapper = shallowMount(Invoice)
    expect(wrapper.is(Invoice)).toBe(true)
  })
  it('should render correct content', () => {
    const wrapper = mount(Invoice)
    expect(wrapper.find('div.card-header > div').text()).toMatch('Invoice')
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Invoice)
    expect(wrapper.element).toMatchSnapshot()
  })
})
