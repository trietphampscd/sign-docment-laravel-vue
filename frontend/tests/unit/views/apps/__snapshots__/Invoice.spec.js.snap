// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invoice.vue renders correctly 1`] = `
<div
  class="animated fadeIn"
>
  <bcard-stub
    bodytag="div"
    footertag="div"
    headertag="div"
    nobody="true"
    subtitletag="h6"
    tag="div"
    titletag="h4"
  >
    <div
      slot="header"
    >
      
      Invoice 
      <strong>
        #90-98792
      </strong>
       
      <a
        class="btn btn-sm btn-info float-right ml-1"
        href="#"
        onclick="javascript:window.print();"
      >
        <i
          class="fa fa-print"
        />
         Print
      </a>
       
      <a
        class="btn btn-sm btn-info float-right"
        href="#"
      >
        <i
          class="fa fa-save"
        />
         Save
      </a>
    </div>
     
    <bcardbody-stub
      bodytag="div"
      subtitletag="h6"
      titletag="h4"
    >
      <brow-stub
        class="mb-4"
        tag="div"
      >
        <bcol-stub
          sm="4"
          tag="div"
        >
          <h6
            class="mb-3"
          >
            From:
          </h6>
           
          <div>
            <strong>
              BootstrapMaster.com
            </strong>
          </div>
           
          <div>
            Konopnickiej 42
          </div>
           
          <div>
            43-190 Mikolow, Poland
          </div>
           
          <div>
            Email: lukasz@bootstrapmaster.com
          </div>
           
          <div>
            Phone: +48 123 456 789
          </div>
        </bcol-stub>
         
        <bcol-stub
          sm="4"
          tag="div"
        >
          <h6
            class="mb-3"
          >
            To:
          </h6>
           
          <div>
            <strong>
              BootstrapMaster.com
            </strong>
          </div>
           
          <div>
            Konopnickiej 42
          </div>
           
          <div>
            43-190 Mikolow, Poland
          </div>
           
          <div>
            Email: lukasz@bootstrapmaster.com
          </div>
           
          <div>
            Phone: +48 123 456 789
          </div>
        </bcol-stub>
         
        <bcol-stub
          sm="4"
          tag="div"
        >
          <h6
            class="mb-3"
          >
            Details:
          </h6>
           
          <div>
            Invoice 
            <strong>
              #90-98792
            </strong>
          </div>
           
          <div>
            March 30, 2013
          </div>
           
          <div>
            VAT: PL9877281777
          </div>
           
          <div>
            Account Name: BootstrapMaster.com
          </div>
           
          <div>
            <strong>
              SWIFT code: 99 8888 7777 6666 5555
            </strong>
          </div>
        </bcol-stub>
      </brow-stub>
       
      <div
        class="table-responsive-sm"
      >
        <table
          class="table table-striped"
        >
          <thead>
            <tr>
              <th
                class="center"
              >
                #
              </th>
               
              <th>
                Item
              </th>
               
              <th>
                Description
              </th>
               
              <th
                class="center"
              >
                Quantity
              </th>
               
              <th
                class="right"
              >
                Unit Cost
              </th>
               
              <th
                class="right"
              >
                Total
              </th>
            </tr>
          </thead>
           
          <tbody>
            <tr>
              <td
                class="center"
              >
                1
              </td>
               
              <td
                class="left"
              >
                Origin License
              </td>
               
              <td
                class="left"
              >
                Extended License
              </td>
               
              <td
                class="center"
              >
                1
              </td>
               
              <td
                class="right"
              >
                $999,00
              </td>
               
              <td
                class="right"
              >
                $999,00
              </td>
            </tr>
             
            <tr>
              <td
                class="center"
              >
                2
              </td>
               
              <td
                class="left"
              >
                Custom Services
              </td>
               
              <td
                class="left"
              >
                Instalation and Customization (cost per hour)
              </td>
               
              <td
                class="center"
              >
                20
              </td>
               
              <td
                class="right"
              >
                $150,00
              </td>
               
              <td
                class="right"
              >
                $3.000,00
              </td>
            </tr>
             
            <tr>
              <td
                class="center"
              >
                3
              </td>
               
              <td
                class="left"
              >
                Hosting
              </td>
               
              <td
                class="left"
              >
                1 year subcription
              </td>
               
              <td
                class="center"
              >
                1
              </td>
               
              <td
                class="right"
              >
                $499,00
              </td>
               
              <td
                class="right"
              >
                $499,00
              </td>
            </tr>
             
            <tr>
              <td
                class="center"
              >
                4
              </td>
               
              <td
                class="left"
              >
                Platinum Support
              </td>
               
              <td
                class="left"
              >
                1 year subcription 24/7
              </td>
               
              <td
                class="center"
              >
                1
              </td>
               
              <td
                class="right"
              >
                $3.999,00
              </td>
               
              <td
                class="right"
              >
                $3.999,00
              </td>
            </tr>
          </tbody>
        </table>
      </div>
       
      <brow-stub
        tag="div"
      >
        <bcol-stub
          lg="4"
          sm="5"
          tag="div"
        >
          
          Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
        
        </bcol-stub>
         
        <bcol-stub
          class="ml-auto"
          lg="4"
          sm="5"
          tag="div"
        >
          <table
            class="table table-clear"
          >
            <tbody>
              <tr>
                <td
                  class="left"
                >
                  <strong>
                    Subtotal
                  </strong>
                </td>
                 
                <td
                  class="right"
                >
                  $8.497,00
                </td>
              </tr>
               
              <tr>
                <td
                  class="left"
                >
                  <strong>
                    Discount (20%)
                  </strong>
                </td>
                 
                <td
                  class="right"
                >
                  $1,699,40
                </td>
              </tr>
               
              <tr>
                <td
                  class="left"
                >
                  <strong>
                    VAT (10%)
                  </strong>
                </td>
                 
                <td
                  class="right"
                >
                  $679,76
                </td>
              </tr>
               
              <tr>
                <td
                  class="left"
                >
                  <strong>
                    Total
                  </strong>
                </td>
                 
                <td
                  class="right"
                >
                  <strong>
                    $7.477,36
                  </strong>
                </td>
              </tr>
            </tbody>
          </table>
           
          <a
            class="btn btn-success"
            href="#"
          >
            <i
              class="fa fa-usd"
            />
             Proceed to Payment
          </a>
        </bcol-stub>
      </brow-stub>
    </bcardbody-stub>
  </bcard-stub>
</div>
`;
